src/js/models/layerstructure.js

```diff
@@ -1089,7 +1089,7 @@ define(function (require, exports, module) {
             }, this);
         }.bind(this));
 
-        return this.mergeDeep({
+        return this.merge({
             layers: nextLayers
         });
```

Comment:
that said, I'd prefer to see this converted to *not* use `withMutations`, which would then get us back to using `merge`.  Maybe that's too drastic a change for now.